### PR TITLE
Deprecate the COMP_xxx API

### PR DESCRIPTION
**Pull request by richsalz**
Fixes: #14506

### CODE REVIEW

**Review by paulidale — DISMISSED** [review]

**Comment by slontis** [comment]
The test failure is relevant

**Review by slontis — COMMENTED** [review]
CHANGES entry?

**Comment by richsalz** [comment]
Update commit pushed.  Let;s see how things look now.

**Comment by richsalz** [comment]
I'm never going to be able to make this work.  I have some kind of block about understanding how openssl does deprecation.  Sorry.  Leaving this around in case someone wants to pick it up and fix it.

File: doc/man3/SSL_COMP_add_compression_method.pod [hunk 1]
```diff
@@ -94,6 +100,8 @@ L<ssl(7)>
 The SSL_COMP_free_compression_methods() function was deprecated in OpenSSL 1.1.0.
 The SSL_COMP_get0_name() and SSL_comp_get_id() functions were added in OpenSSL 1.1.0d.
 
+All functions described here were deprecaed in OpenSSL 3.0.
```
Comment (FdaSilvaYY) [inline doc/man3/SSL_COMP_add_compression_method.pod:103]: typo: deprecaed

File: doc/man3/SSL_COMP_add_compression_method.pod [hunk 2]
```diff
@@ -94,6 +100,8 @@ L<ssl(7)>
...
+
 =head1 COPYRIGHT
 
 Copyright 2001-2020 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (FdaSilvaYY) [inline doc/man3/SSL_COMP_add_compression_method.pod:107]: suggestion: update to 2021...

Comment (t8m) [reply doc/man3/SSL_COMP_add_compression_method.pod:107]: This will get updated automatically before releasing so no problem with that.

**Comment by richsalz** [comment]
can someone clear the "review pending" flag?

**Comment by richsalz** [comment]
Closing this in favor of #15621 and its possible followons.

**Pull request closed without merging**